### PR TITLE
Support for Mir integration

### PR DESCRIPTION
**Pull request by sergefdrv**

### CODE REVIEW

**Comment by sergefdrv** [comment]
@dnkolegov I requested your review just to double-check that everything you need for Mir is supported

**Review by dnkolegov — APPROVED** [review]
LGTM!
From a code review perspective, it seems to me that all functionality we have to have for Mir is here.

File: src/state.rs [hunk 1]
```diff
@@ -114,10 +123,8 @@ impl State {
             && (self.consensus != ConsensusType::Delegated || self.validator_set.len() < 1)
         {
             self.validator_set.push(Validator {
-                subnet: SubnetID::new(&self.parent_id, Address::new_id(sdk::message::receiver())),
                 addr: addr.clone(),
-                // FIXME: Receive address in params
-                net_addr: String::new(),
+                net_addr: String::from(net_addr),
```
Comment (adlrocha) [inline src/state.rs:127]: We should consider checking if this is a multiaddr. We should only support multiaddr for this. We can add a TODO and then consider it for the integration.

**Review by adlrocha — APPROVED** [review]
LGTM. Feel free to merge. Thanks!

**Pull request merged**